platform/gtk4/cog-platform-gtk4.c

```diff
@@ -424,7 +424,13 @@ on_entry_activated(GtkWidget* widget, gpointer user_data)
     struct platform_window* win = user_data;
     GtkEntryBuffer* buffer = gtk_entry_get_buffer(GTK_ENTRY(win->url_entry));
     const gchar* uri = gtk_entry_buffer_get_text(buffer);
-    webkit_web_view_load_uri(win->web_view, uri);
+
+    if (!g_uri_peek_scheme(uri)) {
+        g_autofree gchar *https_uri = g_strconcat("https://", uri, NULL);
+        webkit_web_view_load_uri(win->web_view, https_uri);
+    } else {
+        webkit_web_view_load_uri(win->web_view, uri);
+    }
```

Comment:
Instead of this, you will want to use [cog_uri_guess_from_user_input()](https://igalia.github.io/cog/func.uri_guess_from_user_input.html) unconditionally on the text entered in the text field.

Reply:
@TingPing Could you change the PR to use the suggested function? Thanks!